### ISSUE TITLE
Solver() Psychology ....

### ISSUE THREAD

**Issue by GuyBenhaim** [google/or-tools#797]
Hello,
In many cases, the Solver() makes brilliant optimization choices (!) however, 
in this case, it **consistently provides a solution with a very high cost** (due to ArcCost), even when the penalties assigned to dropping Orders are very low or even 0.

Very high ArcCost are assigned for traveling from the Start or Endpoint to the Nodes to be served and very low cost between Nodes. 
Still, Solver provides a solution that uses these costly segments, where I would e**xpect it to Drop all the Nodes and suffer a much lower penalty**.

Any other factor that might affect such decisions?

**Comment by Mizux** [comment]
Not sure to well understand the issue. I think the start and end node are mandatory i.e. imposed to the solver -> vehicle can't start from any node.

If you want this feature please take a look at https://developers.google.com/optimization/routing/routing_tasks#allowing-arbitrary-start-and-end-locations
tldr create dummy/virtual node with cost 0 to any node, (pro tips: you can also avoid to display it in the solution printer)

**Comment by GuyBenhaim** [comment]
Thx.
Sure, but I would expect the vehicle not to start at all and take the 'hit' of the (low) Drop Penalty X # of Orders.
In my case, it does perform the route, and gets hit by very high ArcCost

**Comment by Mizux** [comment]
I guess it's mandatory for each vehicles to reach its end node even if they don't visit any nodes.

Question: does the cost from Start Node to End Node also expensive ?

**Comment by GuyBenhaim** [comment]
These are two separate Nodes, but at the same location - so 0 cost between these.
Checked some more and this happens also with low costs, there are cases that solver prefers to solve and provides a total cost (for example ~500) rather than skip and have 0 cost. 
It does skip all Nodes in some cases... will try to understand what is the difference that can cause such behavior.

BTW: In one case the solver chooses to serve only the first order (index=0 in locations array), but looking at the CostCallback (breakpoint) it seems that the indexes sent to it are never 0. (Run(int first_index, int second_index)).
**How can it choose order 0 without ever querying the cost to it, or from it?**
